### PR TITLE
PEP 8 Refactoring

### PR DESCRIPTION
**Pull request by Grashalmbeisser**
I took the time to make a big pep8 refactoring. Uses pycodestyle

### CODE REVIEW

File: deploy/modules/__init__.py [hunk 1]
```diff
@@ -2,45 +2,50 @@
...
 import viur.core.prototypes.basic
 
+########################################
+# Manual imports can also be done here!#
+########################################
+
+# noinspection PyUnresolvedReferences
+from viur.core.modules.site import Site as s
+
+# start of script
```
Comment (sveneberth) [inline deploy/modules/__init__.py:21]: The comments here don't make sense like that.... They are placed incorrectly

File: deploy/skeletons/__init__.py [hunk 2]
```diff
@@ -6,15 +6,15 @@
...
-		logging.error("Unable to import skeleton %s" % skelModule[:-3])
+    try:
+        __import__(skelModule[:-3], globals(), locals(), level=1)
+    except ImportError:
+        logging.error("Unable to import skeleton %s" % skelModule[:-3])
 
-	except:
-		raise
+    except Exception as e:
+        raise e
```
Comment (sveneberth) [inline deploy/skeletons/__init__.py:18]: This except is anyway superfluous.
```suggestion
```

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
+[pycodestyle]
+show-pep8 = False
+show-source = True
```
Comment (sveneberth) [inline tox.ini:4]: This was already [in the core a problem](https://github.com/viur-framework/viur-core/blob/130354fdc5851169245d6f0abb1dd41ed3ccd718/tox.ini#L2-L3), please keep it disabled by default
```suggestion
show-source = False
```

File: clean-base.py [hunk 4]
```diff
@@ -1,63 +1,74 @@
...
 VI_VERSION = "3.0.20"
 
 try:
-	whoami = getpass.getuser()
-except:
-	whoami = "viur"
+    whoami = getpass.getuser()
+except Exception as e:
+    whoami = "viur"
+    raise e
```
Comment (sveneberth) [inline clean-base.py:21]: Why?
```suggestion
```

File: clean-base.py [hunk 5]
```diff
@@ -69,34 +80,35 @@
...
-	print(subprocess.check_output("git branch -D main", shell=True).decode().rstrip("\n"))
-	subprocess.check_output("git branch -m main", shell=True)
-	print("Current branch is:", subprocess.check_output("git branch --show-current", shell=True).decode().rstrip("\n"))
-	print("---")
+    print("Clean git history")
+    subprocess.check_output("git checkout --orphan main_tmp", shell=True)
+    print(subprocess.check_output("git branch -D main", shell=True).decode().rstrip("\n"))
+    subprocess.check_output("git branch -m main", shell=True)
+    print(
+        "Current branch is:", subprocess.check_output("git branch --show-current", shell=True).decode().rstrip("\n"))
```
Comment (sveneberth) [inline clean-base.py:110]: This looks very ugly, can we write instead?
```suggestion
	print("Current branch is:",
		  subprocess.check_output("git branch --show-current", shell=True)
		  .decode().rstrip("\n"))
```

File: deploy/main.py [hunk 6]
```diff
@@ -31,9 +31,10 @@
 # General configuration
 #
 
-conf["viur.validApplicationIDs"] = [
-	"{{app_id}}"
-]
+conf["viur.validApplicationIDs"] = \
+    [
+        "{{app_id}}"
+    ]
```
Comment (sveneberth) [inline deploy/main.py:37]: I don't see a problem with the old code. The `\` often does not look really nice.

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Review by phorward — DISMISSED** [review]
Please fix the fixes!

File: clean-base.py (hunk 5 above)
Comment (Grashalmbeisser) [reply clean-base.py:110]: We could, but the PEP8 131 convention needs to be considered. 
So the compromise i can offer is: 
`print(
        "Current branch is:",
        subprocess.check_output("git branch --show-current", shell=True)
        .decode().rstrip("\n")
    )`

File: clean-base.py [hunk 7]
```diff
@@ -1,63 +1,72 @@
...
+import zipfile
+
 from urllib.request import urlopen
 
 VI_VERSION = "3.0.20"
 
 try:
-	whoami = getpass.getuser()
+    whoami = getpass.getuser()
 except:
```
Comment (sveneberth) [inline clean-base.py:18]: Related to [E722](https://www.flake8rules.com/rules/E722.html), we shouldn't use bare excepts. 

tbh, it's not really clear which are the possible exception ([_an exception is raised_](https://docs.python.org/3/library/getpass.html#:~:text=an%20exception%20is%20raised)) Therefor use `Exception`.

```suggestion
except Exception:
```

File: .github/workflows/pep8.yaml [hunk 8]
```diff
@@ -0,0 +1,40 @@
+name: PEP8 linter
```
Comment (sveneberth) [inline .github/workflows/pep8.yaml:1]: Can we please use the same [version as in the core](https://github.com/viur-framework/viur-core/blob/develop/.github/workflows/pep8.yaml), with the GH annotations?

File: deploy/render/__init__.py [hunk 9]
```diff
@@ -1 +1,15 @@
...
+def isList(render, val):
+    return isinstance(val, list)
+
+
+@html.utils.jinjaGlobalFilter
+def isDict(render, val):
+    return isinstance(val, dict)
+=======
 from viur.core.render import admin, html, json, vi, xml
+>>>>>>> feature/pep8Action
```
Comment (sveneberth) [inline deploy/render/__init__.py:15]: Please resolve the merge conflict

**Review by sveneberth — CHANGES_REQUESTED** [review]

File: deploy/render/__init__.py (hunk 9 above)
Comment (Grashalmbeisser) [reply deploy/render/__init__.py:15]: Ouh, oversaw that one, will do!

File: clean-base.py (hunk 5 above)
Comment (Grashalmbeisser) [reply clean-base.py]: anyways i tried to implement a somehow prettier print statement

File: deploy/render/__init__.py [hunk 10]
```diff
@@ -1 +1,10 @@
 from viur.core.render import admin, html, json, vi, xml
+
+@html.utils.jinjaGlobalFilter
+def isList(render, val):
+    return isinstance(val, list)
+
+
+@html.utils.jinjaGlobalFilter
+def isDict(render, val):
+    return isinstance(val, dict)
```
Comment (sveneberth) [inline deploy/render/__init__.py:10]: Please don't re-add old stuff

```suggestion

@html.utils.jinjaGlobalFilter
def isList(render, val):
    return isinstance(val, list)


@html.utils.jinjaGlobalFilter
def isDict(render, val):
    return isinstance(val, dict)
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Review by sveneberth — APPROVED** [review]

**Pull request merged**